### ISSUE TITLE
Undo in puzzles

### ISSUE THREAD

**Issue by Endaris** [panel-attack/panel-game#144]
Suggested by Miz in #post-custom-puzzles
Move puzzles should have an input to revert the last move.
Puzzles generally should have an input to reset to the initial state of the puzzle so the player doesn't have to actively waste all remaining moves or wait for remaining panels / garbage to pop (depending on the puzzle type) if they realise they maneuvered the puzzle into a situation where it is no longer solvable.

Related issue: panel-attack/panel-game#5 Implement replay controls, similar topic of rewinding actions / inputs.

**Comment by JamesVanBoxtel** [comment]
We actually already have a reset, so just need undo

**Comment by Endaris** [comment]
If we have reset already (I was unaware, lol) we should probably indicate that it exists below the move count or something along with the keybind for the active configuration
"Press taunt (keybind in brackets) to reset the puzzle to its initial state"

**Comment by JamesVanBoxtel** [comment]
Yes in general we need on screen prompts for buttons that are non obvious.
